src/components/FeaturePanel/renderers/openingHours/complex.ts

```diff
@@ -36,8 +55,12 @@ export const parseComplexOpeningHours = (
   oneWeekLater.setDate(oneWeekLater.getDate() + 7);
 
   const intervals = oh.getOpenIntervals(today, oneWeekLater);
+  const splittedIntervals = intervals.flatMap(([openingDate, endDate]) =>
+    splitDateRangeAtMidnight([openingDate, endDate]),
```

Comment:
ad splitting interval at midnight) Are you really sure, it is better? It feels natural for me to read an opening hour like `Monday 17-02`, I think it is clear that is over the midnight. Why did you really choose to split it into two?

I would guess the reason could be to compute the "closes soon" more properly? If this is the case, I would propose to still show the "human version" and just compute the opening by the split time. 
Maybe you have some other reason, if so, please tell 🙂 

I added some screenshots to your opening post. Especially in English it is in my opinion unreadable.

Reply:
Interesting view on it, which I agree with for your examples!
I primarily thought about cases where one interval starts at one day and ends at least two days later (e.g. `24/7`). Previously it would show only one day as opened and all the other ones as closed - now each day gets exactly what's correct for the day.

| Old                                                                                     | New                                                                                     |
| --------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------- |
| ![Old](https://github.com/user-attachments/assets/7f27f2fd-2a85-494d-8cc2-805da8e7c503) | ![New](https://github.com/user-attachments/assets/5a1ddac9-b4e8-412e-adcf-c59f59dd9a1c) |

Reply:
I see. Thats why you decided to split it by midnight. Now it makes sense 🙂 Please if possible, (next time) try to add more examples to the PR description, there is so much variation in OSM data.

So – I agree, that your fix to the `24/7` hours is defintely useful, though I still think you should not change the `18-02` to the split version. Do you agree? Can you find a way how to code it? If you like, we may discuss it over discord/videocall 🙂

Reply:
Also one another idea – the `00-00` is rather unclear to me. It was like this before, so feel free to leave it alone, or leave it to a followup. Though, when we are about to show `00-00`, I would strongly prefer `00-24` – what do you think? 🙂

Reply:
### Formatting Enhancements
- Added German and English translations for "midnight" (i.e., `24`/`12 PM`).
- Added German and English translations for "all day" (i.e., `24 hours`).

### Splitting Logic Update
- The system now only splits intervals if they extend beyond 5 AM on the following day. This means that:
  - Intervals like `24/7` will be split.
  - Intervals like `18-02` will remain unchanged.

I chose 5 AM as the split point to balance clarity and usability. It ensures that intervals that extend slightly into the next day remain unaltered while making it clear when an interval overlaps into the morning or midday.

Reply:
Here are some examples:

- [way/649565814](https://osmapp-61a3hbsv2-osm-app-team.vercel.app/way/649565814)
- [node/7101244895](https://osmapp-61a3hbsv2-osm-app-team.vercel.app/node/7101244895)
- [node/11392853254](https://osmapp-61a3hbsv2-osm-app-team.vercel.app/node/11392853254)

Reply:
Thanks. Translation of `00-00` to `24 hours` is a good solution.

Though, please do not translate midnight in other intervals. When it is `18-00`, it is clear already, and adding this to translations will brake all the languages that are not translated yet (becaues english would be used as default). What is even worse is, that the user can change locale to the other one, and it is not really 1:1 to the language then 😅 

After you fix this, feel free to merge it.

btw, I prefixed the PR title with "FeaturePanel:" – please try to add the folder / bigger component where is your PR located. It helps me creating changelogs.